cmake/tools.cmake

```diff
@@ -53,14 +53,17 @@ list (GET COMPILER_VERSION_LIST 0 COMPILER_VERSION_MAJOR)
 # Example values: `lld-10`, `gold`.
 option (LINKER_NAME "Linker name or full path")
 
-if (NOT LINKER_NAME)
-    if (COMPILER_GCC)
-        find_program (LLD_PATH NAMES "ld.lld")
-        find_program (GOLD_PATH NAMES "ld.gold")
-    elseif (COMPILER_CLANG)
-        find_program (LLD_PATH NAMES "ld.lld-${COMPILER_VERSION_MAJOR}" "lld-${COMPILER_VERSION_MAJOR}" "ld.lld" "lld")
-        find_program (GOLD_PATH NAMES "ld.gold" "gold")
-    endif ()
+# s390x doesnt support lld
+if (NOT ARCH_S390X)
```

Comment:
A smaller scope would be nice.

```cmake
       if (COMPILER_GCC)
          if (NOT ARCH_S390X))
              find_program (LLD_PATH NAMES "ld.lld")
          endif()
          find_program (GOLD_PATH NAMES "ld.gold")
       elseif (COMPILER_CLANG)
          if (NOT ARCH_S390X))
              find_program (LLD_PATH NAMES "ld.lld-${COMPILER_VERSION_MAJOR}" "lld-${COMPILER_VERSION_MAJOR}" "ld.lld" "lld")
          endif()
          find_program (GOLD_PATH NAMES "ld.gold" "gold")
       endif ()
```

Note that all CI builds of ClickHouse use lld as a linker. Other linkers (gold, ld) are untested and may be arbitrarily broken (unfortunately - but it is a question of testing costs). So consider making lld work also on s390/x.